src/dependencies/dss-direct-deposit/D3MInit.sol

```diff
@@ -0,0 +1,326 @@
+
+        pool.file("king", aaveCfg.king);
+    }
+
+    function initCompoundPool(
+        DssInstance memory dss,
+        D3MInstance memory d3m,
+        D3MCommonConfig memory cfg,
+        D3MCompoundPoolConfig memory compoundCfg
+    ) internal {
+        D3MCompoundPoolLike pool = D3MCompoundPoolLike(d3m.pool);
+
+        // Sanity checks
+        require(pool.hub() == cfg.hub, "Pool hub mismatch");
+        require(pool.ilk() == cfg.ilk, "Pool ilk mismatch");
+        require(pool.vat() == address(dss.vat), "Pool vat mismatch");
+        require(pool.dai() == address(dss.dai), "Pool dai mismatch");
+        require(pool.comptroller() == compoundCfg.comptroller, "Pool comptroller mismatch");
+        require(pool.comp() == compoundCfg.comp, "Pool comp mismatch");
+        require(pool.cDai() == compoundCfg.cdai, "Pool cDai mismatch");
+
+        pool.file("king", compoundCfg.king);
+    }
+
+    function initAaveRateTargetPlan(
+        D3MInstance memory d3m,
+        D3MAaveRateTargetPlanConfig memory aaveCfg
+    ) internal {
+        D3MAaveRateTargetPlanLike plan = D3MAaveRateTargetPlanLike(d3m.plan);
+        ADaiLike adai = ADaiLike(aaveCfg.adai);
+
+        // Sanity checks
+        require(plan.adai() == address(adai), "Plan adai mismatch");
+        require(plan.stableDebt() == aaveCfg.stableDebt, "Plan stableDebt mismatch");
+        require(plan.variableDebt() == aaveCfg.variableDebt, "Plan variableDebt mismatch");
+        require(plan.tack() == aaveCfg.tack, "Plan tack mismatch");
+        require(plan.adaiRevision() == aaveCfg.adaiRevision, "Plan adaiRevision mismatch");
+        require(adai.ATOKEN_REVISION() == aaveCfg.adaiRevision, "ADai adaiRevision mismatch");
+
+        plan.file("bar", aaveCfg.bar);
+    }
+
+    function initAaveBufferPlan(
+        D3MInstance memory d3m,
+        D3MAaveBufferPlanConfig memory aaveCfg
+    ) internal {
+        D3MAaveBufferPlanLike plan = D3MAaveBufferPlanLike(d3m.plan);
+        ADaiLike adai = ADaiLike(aaveCfg.adai);
+
+        // Sanity checks
+        require(plan.adai() == address(adai), "Plan adai mismatch");
+
+        plan.file("buffer", aaveCfg.buffer);
+    }
+
+    function initCompoundRateTargetPlan(
+        D3MInstance memory d3m,
+        D3MCompoundRateTargetPlanConfig memory compoundCfg
+    ) internal {
+        D3MCompoundRateTargetPlanLike plan = D3MCompoundRateTargetPlanLike(d3m.plan);
+        CDaiLike cdai = CDaiLike(compoundCfg.cdai);
+
+        // Sanity checks
+        require(plan.tacks(compoundCfg.tack) == 1, "Plan tack mismatch");
+        require(cdai.interestRateModel() == compoundCfg.tack, "CDai tack mismatch");
+        require(plan.delegates(compoundCfg.delegate) == 1, "Plan delegate mismatch");
+        require(cdai.implementation() == compoundCfg.delegate, "CDai delegate mismatch");
+        require(plan.cDai() == address(cdai), "Plan cDai mismatch");
+
+        plan.file("barb", compoundCfg.barb);
+    }
```

Comment:
This function apparently is not used. Could potentially be removed.

Reply:
I see, but probably removing unused code doesn't invalidate reviews/audits, does it?

Reply:
Yeah as @gbalabasquer said we want to do exact copy + paste from repos to have auditing be done well in advance along side the relevant repos.

Reply:
> I see, but probably removing unused code doesn't invalidate reviews/audits, does it?

It doesn't but requires the reviewer spends more time checking the diff and removes how trivial is to check it when they are exactly the same.